doc/source/guide/in_depth/code/functional_indepth_example.py

```diff
@@ -0,0 +1,124 @@
+import odl
+
+
+# Here we define the functional
+class MyFunctional(odl.solvers.Functional):
+
+    """This is my functional: ``||x||_2^2 + <x, y>``."""
+
+    def __init__(self, space, linear_term):
+        """Initialize a new instance."""
+        # This comand calls the init of Functional and sets a number of
+        # parameters associated with a functional. All but domain have default
+        # values if not set.
+        super().__init__(space=space, linear=False, grad_lipschitz=2)
```

Comment:
Well depends on your definition of popular, perhaps widely used is a better description.